user_sync/rules.py

```diff
@@ -452,6 +467,32 @@ def sync_umapi_users(self, umapi_connectors):
                         self.updated_user_keys.add(user_key)
                     self.create_umapi_user(user_key, groups_to_add, umapi_info, umapi_connector)
 
+    def create_umapi_groups(self, umapi_connectors):
+        """
+        This is where we create user-groups. If auto_create is enabled,
+        this will pull user-groups from console and compare with mapped_groups. If mapped group does exist
+        in the console, then it will create. Note: Push Mode is not supported
+        :type umapi_connectors: UmapiConnectors
+        """
+        if not self.push_umapi:
+            umapi_info, umapi_connector = self.get_umapi_info(
+                PRIMARY_UMAPI_NAME), umapi_connectors.get_primary_connector()
+            mapped_groups = umapi_info.get_non_normalize_mapped_groups()
+            # pull all user groups from console
+            on_adobe_groups = [normalize_string(g['groupName']) for g in umapi_connector.get_groups()]
+            # verify if group exist and create
+            auto_create = self.options.get('auto_create', None)
+            if auto_create:
+                for mapped_group in mapped_groups:
+                    if normalize_string(mapped_group) not in on_adobe_groups:
+                        self.logger.info("Auto create user-group enabled: Creating %s" % mapped_group)
+                        try:
+                            # create group
+                            res = umapi_connector.create_group(mapped_group)
```

Comment:
see comment above: you should pass the source group name in to create_group so the group description can say what source group it comes from.

Reply:
See my previous comment

Reply:
and mine :)